pallets/dapp-staking/src/lib.rs

```diff
@@ -1399,6 +1408,28 @@ pub mod pallet {
             .into())
         }
 
+        /// Used to set static tier parameters, which are used to calculate tier configuration.
+        /// Tier configuration defines tier entry threshold values, number of slots, and reward portions.
+        ///
+        /// This is a delicate call and great care should be taken when changing these
+        /// values since it has a significant impact on the reward system.
+        #[pallet::call_index(22)]
+        #[pallet::weight(T::WeightInfo::set_static_tier_params())]
+        pub fn set_static_tier_params(
+            origin: OriginFor<T>,
+            params: TierParameters<T::NumberOfTiers>,
+        ) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
```

Comment:
shouldn't we verify the signature first before doing any reads?

Reply:
It has to be root, hence `ensure_root`. Or do you mean something else?

Reply:
I mean ensure_root (signature) first the check if it's enabled